Library/Formula/telegraf.rb

```diff
@@ -0,0 +1,145 @@
+      :revision => "cbeaeb16a013161a98496fad62933b1d21786672"
+  end
+
+  go_resource "github.com/hashicorp/raft" do
+    url "https://github.com/hashicorp/raft.git",
+      :revision => "379e28eb5a538707eae7a97ecc60846821217f27"
+  end
+
+  go_resource "github.com/hashicorp/raft-boltdb" do
+    url "https://github.com/hashicorp/raft-boltdb.git",
+      :revision => "d1e82c1ec3f15ee991f7cc7ffd5b67ff6f5bbaee"
+  end
+
+  go_resource "github.com/hashicorp/go-msgpack" do
+    url "https://github.com/hashicorp/go-msgpack.git",
+      :revision => "fa3f63826f7c23912c15263591e65d54d080b458"
+  end
+
+  go_resource "github.com/golang/protobuf" do
+    url "https://github.com/golang/protobuf.git",
+      :revision => "aece6fb931241ad332956db4f62798dfbea944b3"
+  end
+
+  go_resource "github.com/gogo/protobuf" do
+    url "https://github.com/gogo/protobuf.git",
+      :revision => "05b9dd05e2149cf01fd645b3dc550542556ddea7"
+  end
+
+  go_resource "github.com/boltdb/bolt" do
+    url "https://github.com/boltdb/bolt.git",
+      :revision => "04a3e85793043e76d41164037d0d7f9d53eecae3"
+  end
+
+  go_resource "github.com/armon/go-metrics" do
+    url "https://github.com/armon/go-metrics.git",
+      :revision => "b2d95e5291cdbc26997d1301a5e467ecbb240e25"
+  end
+
+  def install
+    ENV["GOBIN"] = bin
+    ENV["GOPATH"] = buildpath
+    ENV["GOHOME"] = buildpath
+
+    telegraf_path = buildpath/"src/github.com/influxdb/telegraf"
+    telegraf_path.install Dir["*"]
+
+    Language::Go.stage_deps resources, buildpath/"src"
+
+    cd telegraf_path do
+      system "go", "build", "-ldflags", "-X main.Version 0.1.1 -X main.Commit 39c90dd879ce51da60102e9f5694474933607c8f", "cmd/telegraf/telegraf.go"
+    end
+
+    bin.install telegraf_path/"telegraf"
+    etc.install telegraf_path/"etc/config.sample.toml" => "telegraf.conf"
+  end
+
+  plist_options :manual => "telegraf -config #{HOMEBREW_PREFIX}/etc/telegraf.conf"
+
+  def plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>KeepAlive</key>
+        <dict>
+          <key>SuccessfulExit</key>
+          <false/>
+        </dict>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>ProgramArguments</key>
+        <array>
+          <string>#{opt_bin}/telegraf</string>
+          <string>-config</string>
+          <string>#{HOMEBREW_PREFIX}/etc/telegraf.conf</string>
+        </array>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>WorkingDirectory</key>
+        <string>#{var}</string>
+        <key>StandardErrorPath</key>
+        <string>#{var}/log/telegraf.log</string>
+        <key>StandardOutPath</key>
+        <string>#{var}/log/telegraf.log</string>
+      </dict>
+    </plist>
+    EOS
+  end
+
+  test do
+    assert shell_output("telegraf -version").include?("InfluxDB Telegraf")
```

Comment:
Can this test be modified to do something more substantial than e.g. `--version` or `--help`? See `cmake.rb` for an example of an application formula with a good test and `tinyxml2.rb` for an example of a library formula with a good test. Thanks!